clientv3/watch.go

```diff
@@ -61,6 +61,11 @@ func (wr *WatchResponse) Err() error {
 	return nil
 }
 
+// IsProgressNotify returns true if the WatchResponse is progress notification.
+func (wr *WatchResponse) IsProgressNotify() bool {
+	return len(wr.Events) == 0 && !wr.Canceled
```

Comment:
@heyitsanthony 
What's the ordering? Get `Canceled` first and then get closed channel? Or the reverse? Or simultaneously?